src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/Hyperspace.cs

```diff
@@ -70,11 +70,37 @@ public Hyperspace(SparkSession spark)
         public void VacuumIndex(string indexName) => _jvmObject.Invoke("vacuumIndex", indexName);
 
         /// <summary>
-        /// Update indexes for the latest version of the data.
+        /// Update indexes for the latest version of the data. This API provides a few supported refresh
+        /// modes as listed below.
         /// </summary>
         /// <param name="indexName">Name of the index to refresh.</param>
+        /// <param name="mode">Refresh mode. Currently supported modes are <c>incremental</c> and
+        /// <c>full</c>.</param>
         [HyperspaceSince(HyperspaceVersions.V0_0_1)]
-        public void RefreshIndex(string indexName) => _jvmObject.Invoke("refreshIndex", indexName);
+        public void RefreshIndex(string indexName, string mode = "full") =>
```

Comment:
Can we separate out the API that takes in the mode, which is from 0.3?

Reply:
Sure.  Do you think we should keep `mode` as a `string` in .Net, or make it an `enum`?

Reply:
`string` is fine.

Reply:
Updated the PR with `RefreshIndex` split in 2.